lib/src/main/java/graphql/nadel/NadelSchemas.kt

```diff
@@ -30,10 +31,11 @@ data class NadelSchemas constructor(
         internal var serviceExecutionFactory: ServiceExecutionFactory? = null
 
         // .nadel files
-        internal var overallSchemas = mutableMapOf<String, Reader>()
+        internal var overallSchemaReaders = mutableMapOf<String, Reader>()
 
         // .graphqls files
-        internal var underlyingSchemas = mutableMapOf<String, Reader>()
+        internal var underlyingSchemaReaders = mutableMapOf<String, Reader>()
+        internal var underlyingTypeDefs = mutableMapOf<String, TypeDefinitionRegistry>()
```

Comment:
So we need to allow this because the Gateway parses the `Reader` by itself earlier and because it's a Reader we can't reuse it in two places. So we need to allow the Gateway to supply the parsed `TypeDefinitionRegistry`.

Maybe we should do the same for overall schemas. But not needed for now.